src/tests/Microsoft.PowerFx.Core.Tests.Shared/UserDefinedFunctionTests.cs

```diff
@@ -630,6 +630,23 @@ public void TestImperativeUDFParseWithoutSemicolon(string script)
             errors.AddRange(parseResult.Errors ?? Enumerable.Empty<TexlError>());
 
             Assert.True(errors.Count() == 0);
+        }
+
+        [Theory]
+        [InlineData("Count():Number = 1;")]
+        public void TestUDFHasWarningWhenShadowing(string script)
+        {
+            var parserOptions = new ParserOptions()
+            {
+                AllowsSideEffects = true,
+            };
+            var nameResolver = ReadOnlySymbolTable.NewDefault(BuiltinFunctionsCore._library, FormulaType.PrimitiveTypes);
+
+            var parseResult = UserDefinitions.Parse(script, parserOptions);
+            var udfs = UserDefinedFunction.CreateFunctions(parseResult.UDFs.Where(udf => udf.IsParseValid), nameResolver, out var errors);
+            errors.AddRange(parseResult.Errors ?? Enumerable.Empty<TexlError>());
+
+            Assert.True(errors.Any(error => error.MessageKey == "WrnUDF_ShadowingBuiltInFunction" && error.Severity == DocumentErrorSeverity.Warning));
```

Comment:
>True [](http://example.com/codeflow?start=19&length=4)

We should also assert that this is the only error (since that is a "valid" UDF definition)